src/main/solution_space/solution_space_serializer.py

```diff
@@ -18,25 +19,31 @@ class SolutionSpaceSerializer:
     # Returns file path with serialized graph
     @staticmethod
     def serialize(graph: SolutionGraph, serialized_file_prefix: str = 'serialized_graph') -> str:
-        path = os.path.join(SERIALIZED_GRAPH_PATH, f'graph_{graph.id}_{serialized_file_prefix}{EXTENSION.PICKLE.value}')
-        serialize_data_and_write_to_file(path, graph)
-        log.info(f'Graph with id {graph.id} was serialized successfully. File path is {path}')
-        return path
+        graph_name = f'graph_{graph.id}_{serialized_file_prefix}'
+        graph_path = os.path.join(SERIALIZED_GRAPH_PATH, f'{graph_name}{EXTENSION.PICKLE.value}')
+        objects_to_serialize = (graph, IdCounter._instances, IdCounter._id_item_dict_by_class)
```

Comment:
I am not sure that we should serialize all instances. For example, before serializing process we have 2 users and the last Id will be 2, but we can stop the program and run the program with the deserializing process. And we will have the last Id = 2. Is it the correct case?